torch/_subclasses/meta_utils.py

```diff
@@ -262,6 +262,153 @@ def sym_sizes_strides_storage_offset(
+                        else None
+                    )
+                if attr == "_values":
+                    # convert dynamic_dims from NT dim space -> values dim space
+                    inner_policy = FreshCreateSymbolicPolicy(
+                        dynamic_sizes=(
+                            policy.dynamic_sizes[1:]
+                            if isinstance(policy, FreshCreateSymbolicPolicy)
+                            else [DimDynamic.DYNAMIC] * inner_t.dim()
+                        ),
+                        constraint_sizes=(
+                            policy.constraint_sizes[1:]
+                            if isinstance(policy, FreshCreateSymbolicPolicy)
+                            else [None] * inner_t.dim()
+                        ),
+                    )
+
+                    if dense_base is not None:
+                        # For this case, values should be a symbolic view of base.
+                        # Since both values and base are dense, we can use as_strided().
+                        (
+                            inner_sizes,
+                            inner_strides,
+                            inner_storage_offset,
+                        ) = sym_sizes_strides_storage_offset(
+                            inner_t,
+                            inner_src,
+                            inner_policy=inner_policy,
+                        )
+
+                        transformed_tensors_dict[attr] = dense_base.as_strided(
+                            inner_sizes, inner_strides, inner_storage_offset
+                        )
+                        continue
+                else:
+                    # assume fully dynamic for other metadata (e.g. offsets, lengths)
+                    inner_policy = FreshCreateSymbolicPolicy(
+                        dynamic_sizes=[DimDynamic.DYNAMIC], constraint_sizes=[None]
+                    )
+
+                transformed_tensors_dict[attr] = callback(
+                    lambda: empty_create(
+                        inner_t,
+                        inner_src,
+                        inner_policy=inner_policy,
+                    )
+                )
+
+            # We expect JaggedTensor to have a 'ragged_size' in
+            # its context
+            assert isinstance(ctx, dict)
+            assert "ragged_size" in ctx
+            assert isinstance(t._size[1], torch.SymInt)
+            if orig_shape_env is shape_env:
+                # It's already fake and the shape envs line up, reuse the old size
+                # Do not assert singleton_int; it may already
+                # be a variable
+                ctx["ragged_size"] = t._size[1]
+            else:
+                assert t._size[1].node.singleton_int() is not None
+                # Replace the eager ragged size with our freshly
+                # allocated jagged size that has a source
+                ctx["ragged_size"] = get_symbolic_ragged_size(t, source)
```

Comment:
Dumb q: when do the shape envs line up and when do they not?

Reply:
Something about when fake-ifying a fake NT, you want to bring over the existing symbols idk